service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/cmo/AgreedCMOUploadedEventHandler.java

```diff
@@ -29,21 +29,26 @@
     @EventListener
     public void sendNotificationForAdmin(final AgreedCMOUploaded event) {
         CaseData caseData = event.getCaseData();
+        JudgeAndLegalAdvisor judgeAttendingHearing = event.getHearing().getJudgeAndLegalAdvisor();
 
-        CMOReadyToSealTemplate template = contentProvider.buildTemplate(
-            event.getHearing(),
-            caseData.getId(),
-            caseData.getAllocatedJudge(),
-            caseData.getAllRespondents(),
-            caseData.getFamilyManCaseNumber()
-        );
+        if (judgeAttendingHearing.getJudgeEmailAddress() != null || caseData.hasAllocatedJudgeEmail()) {
```

Comment:
If you have some time, some of this is untested (check sonar), so can add 1 more test to cover it